ambry-protocol/src/test/java/com.github.ambry.protocol/RequestResponseTest.java

```diff
@@ -452,9 +454,11 @@ private void doReplicaMetadataRequestTest(short responseVersionToUse) throws IOE
         deserializedReplicaMetadataResponse.getReplicaMetadataResponseInfoList().size());
     Assert.assertEquals("MsgInfo list size in ReplicaMetadataResponse mismatch ", 1,
         deserializedReplicaMetadataResponse.getReplicaMetadataResponseInfoList().get(0).getMessageInfoList().size());
+    Assert.assertEquals("Total size of messages not as expected", size,
+        deserializedReplicaMetadataResponse.getReplicaMetadataResponseInfoList().get(0).getTotalSizeOfAllMessages());
     MessageInfo msgInfo =
         deserializedReplicaMetadataResponse.getReplicaMetadataResponseInfoList().get(0).getMessageInfoList().get(0);
-    Assert.assertEquals("MsgInfo size mismatch ", 1000, msgInfo.getSize());
+    Assert.assertEquals("MsgInfo size mismatch ", size, msgInfo.getSize());
```

Comment:
Can you also call toString() on the response and verify that the length is bounded?

Reply:
I upgraded the test to be better and added this check too